From 95b17c4c4309cfee7fc0fc180dc9133195579c8b Mon Sep 17 00:00:00 2001
From: Cal Archer <carcher002@gmail.com>
Date: Sun, 17 Jul 2016 21:11:30 +0200
Subject: [PATCH] Bluetooth: fix stuttering audio

commit 2772d76990d8424ef3c629dba164fc7111b37db3 caused
a regression in A2DP for Broadcom Bluetooth chipsets
by not sending the HCI_BRCM_SET_ACL_PRIORITY command.

Change this logic so the ACL priority adjustment occurs
unconditionally for non-Broadcom chipsets only.

Reproduction steps: Entering Bluetooth settings which
triggers inquiry and makes the device discoverable while
simultaneously using 2.4GHz WiFi.

Change-Id: Ic40dc8d66ae30c71e0ddf1a2eaeb19fee1e87ecb
---
 stack/l2cap/l2c_utils.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/stack/l2cap/l2c_utils.c b/stack/l2cap/l2c_utils.c
index 5bf1155..2c33c71 100644
--- a/stack/l2cap/l2c_utils.c
+++ b/stack/l2cap/l2c_utils.c
@@ -2613,12 +2613,6 @@ BOOLEAN l2cu_set_acl_priority (BD_ADDR bd_addr, UINT8 priority, BOOLEAN reset_af
         return (FALSE);
     }
 
-    if (p_lcb->acl_priority != priority)
-    {
-        p_lcb->acl_priority = priority;
-        l2c_link_adjust_allocation();
-    }
-
     if (BTM_IS_BRCM_CONTROLLER())
     {
         /* Called from above L2CAP through API; send VSC if changed */
@@ -2642,7 +2636,14 @@ BOOLEAN l2cu_set_acl_priority (BD_ADDR bd_addr, UINT8 priority, BOOLEAN reset_af
                 l2c_link_adjust_allocation();
             }
         }
+    } else {
+        if (p_lcb->acl_priority != priority)
+        {
+            p_lcb->acl_priority = priority;
+            l2c_link_adjust_allocation();
+        }
     }
+
     return(TRUE);
 }
 
